plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -284,22 +329,41 @@ public function menu_order( $menu_order ) {
 		// Initialize our custom order array.
 		$woocommerce_menu_order = array();
 
+		// Cache HPOS status check.
+		$is_hpos_enabled = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled();
+
 		// Get the index of our custom separator.
 		$woocommerce_separator = array_search( 'separator-woocommerce', $menu_order, true );
 
 		// Get index of product menu.
 		$woocommerce_product = array_search( 'edit.php?post_type=product', $menu_order, true );
 
+		// Get index of orders menu.
+		$woocommerce_orders = array_search( 'wc-orders', $menu_order, true );
+
+		// Also check for legacy post type orders menu when HPOS is disabled.
+		if ( false === $woocommerce_orders && ! $is_hpos_enabled ) {
+			$woocommerce_orders = array_search( 'edit.php?post_type=shop_order', $menu_order, true );
+		}
+
 		// Loop through menu order and do some rearranging.
-		foreach ( $menu_order as $index => $item ) {
+		foreach ( $menu_order as $item ) {
 
 			if ( 'woocommerce' === $item ) {
 				$woocommerce_menu_order[] = 'separator-woocommerce';
 				$woocommerce_menu_order[] = $item;
+				$woocommerce_menu_order[] = $is_hpos_enabled ? 'wc-orders' : 'edit.php?post_type=shop_order';
 				$woocommerce_menu_order[] = 'edit.php?post_type=product';
-				unset( $menu_order[ $woocommerce_separator ] );
-				unset( $menu_order[ $woocommerce_product ] );
-			} elseif ( ! in_array( $item, array( 'separator-woocommerce' ), true ) ) {
+				if ( false !== $woocommerce_separator ) {
+					unset( $menu_order[ $woocommerce_separator ] );
+				}
+				if ( false !== $woocommerce_orders ) {
+					unset( $menu_order[ $woocommerce_orders ] );
+				}
+				if ( false !== $woocommerce_product ) {
+					unset( $menu_order[ $woocommerce_product ] );
+				}
+			} elseif ( ! in_array( $item, array( 'separator-woocommerce', 'edit.php?post_type=product', 'wc-orders', 'edit.php?post_type=shop_order' ), true ) ) {
```

Comment:
The previous iteration only checked for `separator-woocommerce` - why all the additional path checks?